controllers/component/component_controller.go

```diff
@@ -75,6 +75,8 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 			if err := helpers.RemoveFinalizerFromComponent(r.Client, logger, ctx, component, helpers.ComponentFinalizer); err != nil {
 				return ctrl.Result{}, err
 			}
+			logger.Error(err, "Application referenced in the component not found. Stopping reconcilation of the request.")
+			return ctrl.Result{}, nil
```

Comment:
To be consistent, can we instead call `HandleLoaderError` ([like this](https://github.com/redhat-appstudio/integration-service/blob/01731e8b4213486d457dcd9063a8915d5179f19d/controllers/integrationpipeline/integrationpipeline_controller.go#L99))? so that we can remove the `logger.Error` calls on Lines 78 and 81.

Reply:
oh nice, updated, thanks!

Reply:
@MartinBasti Can you also change the names of the resources to this instead?

`return helpers.HandleLoaderError(logger, err, "Application", "Component")`

Reply:
yes, I'm stupid I wanted to do it but my attention went somewhere else

I shouldn't be allowed to write code on mondays

Reply:
:laughing:  Thanks for the fix!